docs/tasks/debug-application-cluster/audit.md

```diff
@@ -284,6 +298,138 @@ Events are POSTed as a JSON serialized `EventList`. An example payload:
 }
 ```
 
+### Log Collector Examples
+
+#### Use fluentd to collect and distribute audit events from log file
+
+[Fluentd][fluentd] is an open source data collector for unified logging layer.
+In this example, we will use fluentd to split audit events by different namespaces.
+Note that this example requries json format output support in Kubernetes 1.8.
+
+1. install [fluentd, fluent-plugin-forest and fluent-plugin-rewrite-tag-filter][fluentd_install_doc] in the kube-apiserver node
+1. create a config file for fluentd
+
+          $ cat <<EOF > /etc/fluentd/config
+          # fluentd conf runs in the same host with kube-apiserver
+          <source>
+              @type tail
+              # audit log path of kube-apiserver
+              path /var/log/audit
+              pos_file /var/log/audit.pos
+              format json
+              time_key time
+              time_format %Y-%m-%dT%H:%M:%S.%N%z
+              tag audit
+          </source>
+          
+          <filter audit>
+              #https://github.com/fluent/fluent-plugin-rewrite-tag-filter/issues/13
+              type record_transformer
+              enable_ruby
+              <record>
+              	namespace ${record["objectRef"].nil? ? "<none>":(record["objectRef"]["namespace"].nil? ?  "<none>":record["objectRef"]["namespace"])}
+              </record>
+          </filter>
+          
+          <match audit>
+              # route audit according to namespace element in context
+              @type rewrite_tag_filter
+              rewriterule1 namespace ^(.+) ${tag}.$1
+          </match>
+          
+          <filter audit.**>
+          	  @type record_transformer
+          	  remove_keys namespace
+          </filter>
+          
+          <match audit.**>
+              @type forest
+              subtype file
+              remove_prefix audit
+              <template>
+                  time_slice_format %Y%m%d%H
+                  compress gz
+                  path /var/log/audit-${tag}.*.log
+                  format json
+                  include_time_key true
+              </template>
+          </match>
+1. start fluentd
+
+          $ fluentd -c /etc/fluentd/config  -vv
+1. start kube-apiserver with the following options:
+
+          --audit-policy-file=/etc/kubernetes/audit-policy.yaml --audit-log-path=/var/log/kube-audit --audit-log-format=json
+1. check audits for different namespaces in /var/log/audit-*.log
+
+#### Use logstash to collect and distribute audit events from webhook backend
+
+[Logstash][logstash] is an open source, server-side data processing tool. In this example,
+we will use logstash to collect audit events from webhook backend, and save events of
+different users into different files.
+
+1. install [logstash][logstash_install_doc]
+1. create config file for logstash
+
+          $ cat <<EOF > /etc/logstash/config
+          input{
+              http{
+                  #TODO, does logstash support https input?
```

Comment:
Yes, but it's disabled by default: https://www.elastic.co/guide/en/logstash/current/plugins-inputs-http.html#plugins-inputs-http-ssl